plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -848,6 +856,48 @@ int amdgpu_plugin_dump_file(int fd, int id)
 
 	}
 
+	for (int i = 0; i < helper_args.num_of_devices; i++) {
+		struct tp_node *dev;
+		int ret_thread = 0;
+		thread_datas[i].gpu_id = devinfo_bucket_ptr[i].actual_gpu_id;
+		thread_datas[i].bo_buckets = bo_bucket_ptr;
+		thread_datas[i].bo_info_test = e->bo_info_test;
+		thread_datas[i].pid = e->pid;
+		thread_datas[i].num_of_bos = helper_args.num_of_bos;
+
+		dev = sys_get_node_by_gpu_id(&src_topology, thread_datas[i].gpu_id);
+		if (!dev) {
+			ret = -EFAULT;
+			goto failed;
+		}
+
+		thread_datas[i].drm_fd = open_drm_render_device(dev->drm_render_minor);
+		if (thread_datas[i].drm_fd < 0) {
+			ret = -EFAULT;
```

Comment:
Doesn't drm_fd already contain an error code? Why not assign that to ret?

Reply:
ACK